geopandas/sindex.py

```diff
@@ -244,26 +241,18 @@ def query_bulk(self, geometry, predicate=None, sort=False):
                 input_geometry_index.extend([i] * len(res))
             return np.vstack([input_geometry_index, tree_index])
 
-        def intersection(self, coordinates, objects=False):
-            """Find tree geometries that intersect the input coordinates.
+        def intersection(self, coordinates):
+            """Wrapper for rtree.index.Index.intersection.
 
             Parameters
             ----------
             coordinates : sequence or array
                 Sequence of the form (min_x, min_y, max_x, max_y)
                 to query a rectangle or (x, y) to query a point.
-            objects : boolean, default False
-                If True, return the label based indexes. If False, integer indexes
-                are returned.
             """
-            if objects:
-                warn(
-                    "`objects` is deprecated and will be removed in a future version. "
-                    "Instead, use `iloc` to index your GeoSeries/GeoDataFrame using "
-                    "integer indexes returned by `intersection`.",
-                    FutureWarning,
-                )
-            return super().intersection(coordinates, objects)
+            return np.array(
+                list(super().intersection(coordinates, objects=False)), dtype=np.intp,
+            )
```

Comment:
This is technically an API change: we now return a numpy array instead of a generator (which is what rtree does). This is just for consistency, but I'm open to putting in a warning, or just holding off altogether (I don't remember why I did this, it may be _required_ for some test).

Reply:
If we do not need to do this, I would keep it as it is now. In theory, this PR should not change anything for the user (apart from `objects` deprecation).

Reply:
I reverted and ran the tests. I didn't encounter issues.